src/visuals/main_menu.rs

```diff
@@ -0,0 +1,216 @@
+                    ..default()
+                })
+                    //Node for the play button
+                    .with_children(|parent|{
+                        parent.spawn((ButtonBundle {
+                            style: Style {
+                                size: Size::new(Val::Px(370.0), Val::Px(95.0)),
+                                justify_content: JustifyContent::Center,
+                                align_items: AlignItems::Center,
+                                ..default()
+                            },
+                            ..default()
+                        },
+                            ButtonTypes::Play,
+                            MainMenuElements::Button,
+                        ))
+                        //Node for play text
+                        .with_children(|parent|{
+                            parent.spawn((TextBundle::from_section (
+                                "Play",
+                                TextStyle {
+                                    font_size: 80.0,
+                                    color: WHITE,
+                                    ..default()
+                                }
+                            ),
+                            MainMenuElements::Text
+                            ));
+                        });
+                    }) 
+                    //Node for the quit button
+                    .with_children(|parent|{
+                        parent.spawn((ButtonBundle {
+                            style: Style {
+                                size: Size::new(Val::Px(370.0), Val::Px(95.0)),
+                                justify_content: JustifyContent::Center,
+                                align_items: AlignItems::Center,
+                                ..default()
+                            },
+                            ..default()
+                        },
+                            ButtonTypes::Quit,
+                            MainMenuElements::Button,
+                        ))
+                            //Node for quit text
+                        .with_children(|parent|{
+                            parent.spawn((TextBundle::from_section (
+                                "Quit",
+                                TextStyle {
+                                    font_size: 80.0,
+                                    color: WHITE,
+                                    ..default()
+                                }
+                            ),
+                            MainMenuElements::Text
+                            ));
+                        });
+                    });
+            });
+}
+
+pub fn load_background_image(
+    asset_server: Res<AssetServer>, 
+    mut element_query: Query<(&MainMenuElements, &mut UiImage), Added<MainMenuElements>>)
+{
+    for (element, mut image) in &mut element_query {
+        
+        if let MainMenuElements::BackgroundImage = element {                
```

Comment:
This is not a suggestion, but an observation.

I've never seen this pattern of comparison, especially with a if-let statement. I had to look this up after doing a double-take.

Apparently you can do a comparison _and_ assign a variable. From this [resource](https://doc.rust-lang.org/reference/expressions/if-expr.html)
```rust
let dish = ("Ham", "Eggs");

if let ("Ham", b) = dish {
    println!("Ham is served with {}", b);
}
```

Reply:
This was a suggestion that was provided when I made a single branch match statement. According to the documentation they're equivalent as shown by this example in your linked resource:

```
An if let expression is equivalent to a [match expression](https://doc.rust-lang.org/reference/expressions/match-expr.html) as follows:


if let PATS = EXPR {
    /* body */
} else {
    /*else */
}
is equivalent to

match EXPR {
    PATS => { /* body */ },
    _ => { /* else */ },    // () if there is no else
}
```